src/OSPSuite.Core/Domain/Services/ParameterIdentifications/ParameterIdentificationRun.cs

```diff
@@ -82,18 +82,25 @@ public class ParameterIdentificationRun : IParameterIdentificationRun
          _jacobianMatrixCalculator = jacobianMatrixCalculator;
       }
 
-      public void InitializeWith(IParameterIdentifcationRunInitializer runInitializer)
+      public void InitializeWith(IParameterIdentificationRunInitializer runInitializer)
       {
          _runInitializer = runInitializer;
       }
 
-      private void initialize()
+      private void initialize(CancellationToken cancellationToken)
       {
-         _parameterIdentification = _runInitializer.InitializeRun().Result;
+         _parameterIdentification = _runInitializer.InitializeRun(cancellationToken).Result;
          RunResult.Description = _parameterIdentification.Description;
          _optimizationAlgorithm = _optimizationAlgorithmMapper.MapFrom(_parameterIdentification.AlgorithmProperties);
          _residualCalculator = _residualCalculatorFactory.CreateFor(_parameterIdentification.Configuration);
-         _parameterIdentification.AllSimulations.Each(s => _allSimModelBatches.Add(s, createSimModelBatch(s)));
+         var parallelOptions = createParallelOptions(_cancellationToken);
+
+         Parallel.ForEach(_parameterIdentification.AllSimulations, parallelOptions, simulation =>
+         {
+            parallelOptions.CancellationToken.ThrowIfCancellationRequested();
+            _allSimModelBatches.TryAdd(simulation, createSimModelBatch(simulation));
```

Comment:
Also initializing simModel in paralellel. When optimizing multiple simulations, this will be a massive gain in startup time